Modules/Segmentation/LevelSets/include/itkParallelSparseFieldLevelSetImageFilter.h

```diff
@@ -712,7 +712,8 @@ class ITK_TEMPLATE_EXPORT ParallelSparseFieldLevelSetImageFilter
 
   /** Thread-specific data */
   std::vector<TimeStepType> m_TimeStepList;
-  std::vector<bool>         m_ValidTimeStepList;
+  // NB: although semantically boolean, int is used because vector<bool> is not thread safe due to the possibility of multiple bits being packed together in the same memory location.
+  std::vector<int>          m_ValidTimeStepList;
```

Comment:
Do I understand correctly that each Boolean value `m_ValidTimeStepList[i]` just indicates whether the corresponding value `m_TimeStepList[i]` is _valid_?

Reply:
I've never used this class, but it looks something like that, yes.

Reply:
Obviously, your PR is a change of the API (or at least the ABI). Would that be a problem to anyone?

And then, if we are willing to change the API for the sake of thread safety, could we possibly improve it even more?

I just tried entirely removing `m_ValidTimeStepList`, and the corresponding `std::vector<bool>` parameter from all `ResolveTimeStep` member functions, and to my surprise, it just worked fine! Even the test `itkImplicitManifoldNormalVectorFilterTest` ran successfully!

Would it be fine to you as well, to simply just remove `m_ValidTimeStepList`?